miner/worker.go

```diff
@@ -924,7 +924,14 @@ func (w *worker) commitNewWork(interrupt *int32, noempty bool, timestamp int64)
 	}
 	// Set baseFee and GasLimit if we are on an EIP-1559 chain
 	if w.chainConfig.IsLondon(header.Number) {
-		header.BaseFee = misc.CalcBaseFee(w.chainConfig, parent.Header())
+		if w.chainConfig.EnableEIP2718 && w.chainConfig.EnableEIP1559 {
+			header.BaseFee = misc.CalcBaseFee(w.chainConfig, parent.Header())
+		} else {
+			// When disabling EIP-2718 or EIP-1559, we do not set baseFeePerGas in RPC response.
+			// Setting BaseFee as 0 here can help outside SDK calculates l2geth's RLP encoding,
+			// otherwise the l2geth's BaseFee is not known from the outside.
```

Comment:
Discussed this with @zimpha , setting BaseFee as 0 to help outside dev add a patch, because if we do not set `baseFeePerGas` in block header, the blockhash will be inconsistent with the l2geth which has `baseFeePerGas` (MetaMask does not have this issue).